dht_net.go

```diff
@@ -76,6 +80,17 @@ func (dht *IpfsDHT) handleNewMessage(s inet.Stream) bool {
 		case nil:
 		}
 
+		startedHandling := time.Now()
+
+		stats.RecordWithTags(
+			ctx,
+			[]tag.Mutator{
```

Comment:
Use `UpsertMessageType` function

Reply:
Better yet, move upserting of the message type to a `tag.New` so that the `ctx` has the message type and then remove the upsert of message type from the latency record statement at the end of the function.